src/pykiso/test_coordinator/test_execution.py

```diff
@@ -128,6 +132,53 @@ def set_skipped(test_case: BasicTest) -> None:
     list(map(set_skipped, filter(is_skip_test, base_suite)))
 
 
+def apply_test_case_filter(
+    all_tests_to_run: unittest.TestSuite,
+    test_class_pattern: str,
+    test_case_pattern: str,
+) -> unittest.TestSuite:
+    """Apply a filter to run only test cases which matches given expression
+
+    :param all_tests_to_run: a dict containing all testsuites and testcases
+    :param test_class_pattern: pattern to select test class as unix filename pattern
+    :param test_case_pattern: pattern to select test case as unix filename pattern
+    :return: new test suite with filtered test cases
+    """
+
+    base_suite = test_suite.flatten(all_tests_to_run)
+
+    def is_active_test(
+        test_case: BasicTest, test_class_pattern: str, test_case_pattern: str
+    ) -> bool:
+        """_summary_
+
+        :param test_case: unittest test
+        :param test_class_pattern: pattern to select test class as unix filename pattern
+        :param test_case_pattern: pattern to select test case as unix filename pattern
+        :return: True if test matches patterns else False
+        """
+        test_class_name = re.sub(r"-\d+-\d+", "", test_case.__class__.__name__)
```

Comment:
Normally this part should not be much cpu consuming because the pattern an the strings to compare with are in our cases very short.

I also think that Regexes in python should be implemented efficiently.